src/AI/AIUtils.java

```diff
@@ -26,19 +26,25 @@
 
 public class AIUtils
 {
+  /** Overload for {@link #getAvailableUnitActions(Unit, GameMap, boolean)} */
```

Comment:
Comment should probably state or `@note` or something that is assumes `includeOccupiedDestinations==false`.

Reply:
Actually, hold up. Why does this need an overload at all? The prior implementation just assumed `includeOccupiedDestinations==true` in `getAvailableUnitActions` -- if the unit can't board transports, those actions won't be found anyway.

Reply:
JOIN is also an action that exists. For a bit there at least, Wally would use it to move units out of the way of other units' attacks.

Reply:
By the way: Eclipse *does* have a function to generate them automatically.
If you enter # while in a javadoc, it will autoprompt/complete signature links for you.

Reply:
Isn't the existence of JOIN just another argument to include occupied destinations when finding unit actions?

Reply:
Right, but I didn't feel like touching all of the invocations and none of our present AIs are actually interested in joining or loading.

Reply:
I'm not understanding something, or maybe you aren't. The function is called `getAvailableUnitActions`, but you have modified it to ignore unit actions that can end on friendly units for no reason I can discern. I don't even know why this function needs an overload to allow specifying that we *don't* want to include those results; it didn't use to have one, and nobody uses it.
The 3-param overload is only ever invoked by the 2-param overload, and the 2-param overload is only ever called from ISAI. "Touching all the invocations" would just involve removing the superfluous overload and changing the other to assume `includeOccupiedDestinations==true`.
I don't want to leave this function returning different results than advertised, when it did what was expected before and nothing in this PR depends on it changing functionality.

Reply:
> I'm not understanding something, or maybe you aren't. The function is called `getAvailableUnitActions`, but you have modified it to ignore unit actions that can end on friendly units for no reason I can discern.

I believe I made the initial modification after observing the AIs get stuck in an infinite loop trying to send a unit to an occupied space. This "use case" is actually not as bad anymore, since the MoveEvent *will* end the offending unit's turn even if the move is invalid (so the game can continue).

> I don't even know why this function needs an overload to allow specifying that we _don't_ want to include those results; it didn't use to have one, and nobody uses it.

In addition to ISAI, it's invoked by `getAvailableUnitActionsByType()`, which is used by Muriel.

> I don't want to leave this function returning different results than advertised, when it did what was expected before and nothing in this PR depends on it changing functionality.

I guess I don't have any strong objections now that an infinite loop of failed Wait actions is no longer possible with a single unit.

Reply:
> AIs get stuck in an infinite loop... MoveEvent will end the offending unit's turn

If AIs are trying to execute `WAIT` commands derived from `JOIN` or `LOAD` actions, then I would think that is the thing to fix, and modifying a MOVE action to always end the unit's turn no matter what is fine I guess, but is a superficial patch fix. That's beyond scope for this PR though, other than to maybe record the issue.
Also, badly-constructed actions should just end the AI's turn altogether (so maybe that's another issue to fix).

> In addition to ISAI, it's invoked by getAvailableUnitActionsByType(), which is used by Muriel.

Touchè, mon frère.